packages/@react-native-windows/cli/src/runWindows/runWindows.ts

```diff
@@ -271,7 +271,11 @@ async function runWindowsInternal(
   // Restore packages.config files for dependencies that don't support PackageReference.
   runWindowsPhase = 'RestorePackagesConfig';
   try {
-    await buildTools.restorePackageConfigs(slnFile);
+    await buildTools.restorePackageConfigs(
+      slnFile,
+      options.arch,
+      deploy.getBuildConfiguration(options),
```

Comment:
Nit: We call this later to get the configuration for the build step (line 323), move earlier and re-use it:

`const buildType = deploy.getBuildConfiguration(options)`;